From 246fc49b718e0548d7a2d5ab095d6b547dbaa532 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 7 Nov 2023 13:40:50 +0100
Subject: [PATCH 0024/1695] plat-stm32mp1: default disable reserved shared
 memory

Changes stm32mp15 default configuration to not embed OP-TEE's reserved
static shared memory. This change is aligned with a recent change in
TF-A [1] that also default disables the related configuration switch.
Note that TF-A/stm32mp1 deprecates this configuration in order to fully
remove its support in the future [2].

Note that when CFG_CORE_RESERVED_SHM is disabled the default 2MB area
is now used by OP-TEE as secure memory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3fa2f054f3ceff09c6558d3d51ab954e3d15f7a1 plat-stm32mp1: default disable reserved shared memory]
Change-Id: I3f3e945f2fa2e25e246c6be62686dbfa0ec8e0f1
Link: https://git.trustedfirmware.org/TF-A/trusted-firmware-a.git/commit/?id=fb1d3bd9330ce70f735a344dd4223faffb261118 [1]
Link: https://git.trustedfirmware.org/TF-A/trusted-firmware-a.git/commit/?id=12e683a68049f6a3d0985a2cd1564e00115e809f [2]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 8b3ac1f6bc0df62a6524bed9f21a4c97bc8fe3ce]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 058199a48..c488c5887 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -129,7 +129,7 @@ $(call force,CFG_SECONDARY_INIT_CNTFRQ,y)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,y)
 $(call force,CFG_STM32_SAES,n)
 $(call force,CFG_STM32MP15_CLK,y)
-CFG_CORE_RESERVED_SHM ?= y
+CFG_CORE_RESERVED_SHM ?= n
 CFG_EXTERNAL_DT ?= y
 CFG_STM32_BSEC_SIP ?= y
 CFG_TEE_CORE_NB_CORE ?= 2
-- 
2.47.2

